zaza/openstack/charm_tests/neutron/tests.py

```diff
@@ -591,6 +592,28 @@ def test_901_pause_and_resume(self):
                                pgrep_full=self.pgrep_full):
             logging.info('Testing pause resume')
 
+    def test_openvswitch_firewall_ovsdb_protocols(self):
+        """Validate activation of OpenFlow protocols on driver change.
+
+        This helps validate the effect and need for workaround for LP: #1852221
+        """
+        if self.current_os_release < self.xenial_mitaka:
+            logging.info('Skipping test for < xenial-mitaka')
+            return
+
+        with self.config_change({'firewall-driver': ''},
+                                {'firewall-driver': 'openvswitch'}):
+            for unit in zaza.model.get_units('neutron-openvswitch',
+                                             model_name=self.model_name):
+                # ``ovs-ofctl`` will throw an error if ovsdb-server refuses to
```

Comment:
@fnordahl I wonder if we should check that it throws an error first and then check that the change fixes the issue.